### PR TITLE
some simple refactoring

### PR DESCRIPTION
**Pull request by ashishnegi**
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/567)

### CODE REVIEW

File: worker/task.go [hunk 1]
```diff
@@ -235,7 +235,7 @@ func processTask(q *task.Query, gid uint32) (*task.Result, error) {
 		for _, uid := range uids.Uids {
 			key := x.DataKey(attr, uid)
 			pl, decr := posting.GetOrCreate(key, gid)
-
+			defer decr()
```
Comment (ashishnegi) [inline worker/task.go]: only potential flaw in this that now posting's reference count will decrease only at function unwinding,
not immediately in each iteration of for loop.
Can that be a concern ? What if uids.Uids are larger ?

File: worker/task.go [hunk 2]
```diff
@@ -244,12 +244,11 @@ func processTask(q *task.Query, gid uint32) (*task.Result, error) {
 				newValue.Val = x.Nilbyte
 			}
 			values = append(values, newValue)
-			decr() // Decrement the reference count of the pl.
 		}
 
 		filtered := types.FilterGeoUids(uids, values, geoQuery)
 		for i := 0; i < len(out.UidMatrix); i++ {
-			out.UidMatrix[i] = algo.IntersectSorted([]*task.List{out.UidMatrix[i], filtered})
+			algo.IntersectWith(out.UidMatrix[i], filtered)
```
Comment (ashishnegi) [inline worker/task.go]: intersectWith mutates the first list ; which is what we want.
It is faster than IntersectSorted. Also no new allocation of list happens.

**Comment by ashwin95r** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[worker/task.go, line 238 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/567#-Kc1D_F3raXzzSUkptzA-r1-238:-Kc1JU4e6L03kLEc6Aho:b6dgzjp) ([raw file](https://github.com/dgraph-io/dgraph/blob/83e4f68975e34fab37768e935425ea262548e400/worker/task.go#L238)):*
<details><summary><i>Previously, ashishnegi (Ashish Negi) wrote…</i></summary><blockquote>

only potential flaw in this that now posting's reference count will decrease only at function unwinding,
not immediately in each iteration of for loop.
Can that be a concern ? What if uids.Uids are larger ?
</blockquote></details>

Yes. So I think its better put at the end of loop. Right?

---

*[worker/task.go, line 251 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/567#-Kc1D_F3raXzzSUkptzA-r1-251:-Kc1JY1g-K04Vwfu3N38:b-vs1pr4) ([raw file](https://github.com/dgraph-io/dgraph/blob/83e4f68975e34fab37768e935425ea262548e400/worker/task.go#L251)):*
<details><summary><i>Previously, ashishnegi (Ashish Negi) wrote…</i></summary><blockquote>

intersectWith mutates the first list ; which is what we want.
It is faster than IntersectSorted. Also no new allocation of list happens.
</blockquote></details>

Looks good.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/dgraph/567)*

**Comment by ashishnegi** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[worker/task.go, line 238 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/567#-Kc1D_F3raXzzSUkptzA-r1-238:-Kc1MUKiOOnD-qXbWw9t:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/83e4f68975e34fab37768e935425ea262548e400/worker/task.go#L238)):*
<details><summary><i>Previously, ashwin95r (Ashwin Ramesh) wrote…</i></summary><blockquote>

Yes. So I think its better put at the end of loop. Right?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/dgraph/567)*

**Comment by ashwin95r** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r1.
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/dgraph/567#-:-Kc1kfohColGE1PvYkgb:bnfp4nl)*

**Pull request closed without merging**